lib/handler/range.c

```diff
@@ -0,0 +1,119 @@
+#include <assert.h>
+#include <stddef.h>
+#include <string.h>
+
+#include "h2o.h"
+
+#if __GNUC__ >= 3
+# define likely(x) __builtin_expect(!!(x), 1)
+# define unlikely(x) __builtin_expect(!!(x), 0)
+#else
+# define likely(x) (x)
+# define unlikely(x) (x)
+#endif
```

Comment:
Please move the definitions to h2o/memory.h with changing there names to `H2O_LIKELY` and `H2O_UNLIKELY`.